### ISSUE TITLE
Accessibility Issues - Listbox role + tabIndex

### ISSUE THREAD

**Issue by Sam-CellarDoor** [express-labs/pure-react-carousel#403]
### Documenting three issues that should be discussed / actioned on:

**1. The carousel should likely not be using a "listbox" role**

This issue [has already been discussed at length here](https://github.com/express-labs/pure-react-carousel/issues/112), but I do not see a clear resolution. I understand there is an argument that the screen reader cannot read the contents of the carousel correctly without the 'listbox' role, and therefore it's better to leave it in. We should challenge this type of thinking when it comes to accessibility -- roles have a very specific meaning that is analogous to semantic HTML. Using a role is like a contract in that we are telling the person using the screen reader that they can expect certain functionality & a certain interaction. 

In this case, we are communicating that there is essentially a list of selectable items (that can contain images, etc...) - which is very rarely the actual functionality of a carousel. Not only that, we are communicating that we have implemented a keyboard control pattern for navigating through this interaction (look [here for the expected keyboard interaction](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role#keyboard_interactions)). Neither of these are the case, so therefore it's better to leave the role empty and allow developers to implement these features themselves. This would mean that 'role' and all 'aria' attributes are available on Slider & Slide.

[This guide](https://www.w3.org/WAI/tutorials/carousels/) demonstrates how to correctly + accessibly create a carousel.
 

**2. Remove tabIndex = 0 on Slider & Slide**

This [article states it best](https://www.a11yproject.com/posts/how-to-use-the-tabindex-attribute/), "there is a myth that applying tabindex="0" to non-interactive elements (paragraphs, headings, lists, etc.) helps “improve” accessibility by providing a way for a person who uses a screen reader to focus on all the content present in a webpage or web app. Unfortunately, this well-intentioned idea actually does not create a good assistive technology experience". Instead, interactive elements (such as forms & links) will be included in the tab order by default, and should be used instead. 

**3. DotGroup considerations**

The DotGroup is comprised of buttons which will automatically be included in the tab order. Developers do not have the ability to easily control whether or not this is the case (ie. control tabIndex on the carousel__dot buttons). 

Also, [this article](https://www.w3.org/WAI/tutorials/carousels/functionality/) demonstrates how the DotGroup could be used to provide intuitive and accessible navigation by populating the buttons with visually hidden titles + aria attributes.

Both of these necessitate manual control over the button array. I believe this feature technically already exists with the 'renderDots' prop, however there may be a more developer friendly way to go about it.

### Suggested implementation:

- Remove "listbox" role from Slider
- Remove "option" role from Slide
- Ensure 'role' and other aria html attributes are available to developers on these components
- Remove tabIndex = 0 from Slider & Slide

### Useful Resources

- https://www.w3.org/WAI/tutorials/carousels/functionality/
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role

### Notes:

Thanks for all the hard work y'all have done on this project! I will likely continue to add to + edit this post as I continue to research the problem.

**Comment by leadegroot** [comment]
Our implementation uses an anchor on each slide of the carousel to allow the user to click through for more information - presumably not an unusual way to do it.
Testing accessibility with cypress and axe-core is currently giving a fail, throwing this error:
https://dequeuniversity.com/rules/axe/4.3/nested-interactive?application=axeAPI
I am assuming at this point that this is due to the above mentioned listbox-option roles
